sqlitecluster/SQLiteNode.cpp

```diff
@@ -561,14 +564,13 @@ bool SQLiteNode::update() {
         }
 
         // No leader and we're in sync, perhaps everybody is waiting for us
-        // to stand up?  If we're higher than the highest priority, and are
-        // connected to enough full peers to achieve quorum we should be
-        // leader.
+        // to stand up?  If we're higher than the highest priority, are using 
+        // a real priority and are not a permafollower, and are connected to 
+        // enough full peers to achieve quorum, we should be leader.
         if (!currentLeader && numLoggedInFullPeers * 2 >= numFullPeers &&
-            _priority > highestPriorityPeer->calc("Priority")) {
+            _priority > 0 && _priority > highestPriorityPeer->calc("Priority")) {
```

Comment:
If you're not checking for `_priority > 0`, a priority 0 permaslave in waiting will stand up if the priorities of the full peers are -1 because they haven't switched to using their real priority.

Reply:
If `_priority > 0` is true then `_priority` is true, right?

Reply:
Yes, I can't read.

Can `highestPriorityPeer->calc("Priority")` be less than 0?

Reply:
It can be less than zero, because of the "-1" fake priority for hosts that haven't reached WAITING yet.

> Would that make numLoggedInFullPeers * 2 >= numFullPeers fail?
Not sure, I will check. I think there's something about "assuming they're logged in if they aren't connected" or something. I'll dig in and get back to you.

Reply:
The way we set `numLoggedInFullPeers` is by looking to see if the peer is logged in. `SIEquals((*peer)["LoggedIn"], "true")` This has a race condition, right, if you lose connection with half your peers at the same time?

Reply:
`numLoggedInFullPeers` first excludes permafollowers (based on configured peerList) and then depends on the logged in state:
```
            // Make sure we're a full peer
            if (peer->params["Permafollower"] != "true") {
                // Verify we're logged in
                ++numFullPeers;
                if (SIEquals((*peer)["LoggedIn"], "true")) {
                    // Verify we're still fresh
                    ++numLoggedInFullPeers;
```

so, any node can be logged in, but still advertising the "fake" priority.